### DISCUSSION TITLE
A suitable way to expand cluster space

### DISCUSSION THREAD

**Discussion by HwiLu** [ClickHouse/ClickHouse#60991] [Q&A] [answered]
Clickhouse version:  `22.8.15.23 (official build)`

I want to expand the capacity of a cluster with two replicas and one shard. 

The data disk space of this cluster is almost full.  Which expansion plan is better?

Our cluster uses replicated tables, not distributed table, so I plan to stop the service and add new disks, in this case the storage space of each node becomes larger.

I guess it should be useless to add new nodes because the total space will not increase when we are still 2 copies and 1 shard.

Is it possible to divide it into multiple shards?  And then we need to use distributed tables, Does this require major business changes?

We may add one or two new nodes and they will be in different computer rooms or just add some disk .

Thank you 😄

**Comment by shelly-ctrl** [comment, marked as answer]
It is recommended to expand the disk space first. After the expansion, you can also consider adding new nodes to further improve data writing performance.

**Reply by HwiLu** [reply]
Good advice.
